File: php/problem3.php
<?php
//solution: 6857
echo eulerProblem3() . "\n";

function eulerProblem3() {
    $max = 1;
    $limit = 600851475143;
    for($factor = 2; $factor <= $limit; $factor = nextPrime($factor)) {
        if($limit % $factor == 0) {
            $max = $factor;
            $limit = $limit / $factor;
        }
    }
    return $max;
}

function nextPrime($n) {
    $n = $n + 1;
    while($n++) {
        if(isPrime($n)) {
            return $n;
        }
    }
}

function isPrime($n) {
    for($i = 2; $i < $n; ++$i) {
        if($n % $i == 0) {
            return 0;
        }
    }
    return 1;
}

?>

File: python/problem3.py
#!/usr/bin/python
#solution: 6857

import math

def is_prime(n):
    if n == 2:
        return True
    if n % 2 == 0:
        return False
    i = 3
    while i < math.sqrt(n):
        if n % i == 0:
            return False
        i += 2
    return True
    

def euler_problem_3():
    i, max, limit = 3, 0, 600851475143
    while i < math.sqrt(limit):
        if is_prime(i) and limit % i == 0:
            max = i
        i += 2
    return max

print euler_problem_3()
